src/Processor/WordProcessor.hs

```diff
@@ -0,0 +1,107 @@
+module Processor.WordProcessor (calculateDiff, getWordsByWordDiffList, getWordDiffListByWords, findBestWord) where
+
+import Control.Monad.Trans.State (State, evalState, get, modify)
+import Data.Char (toLower)
+import Data.GameState (Color (..), WordDiff (..))
+import Data.Map.Strict (adjust, (!?), union)
+import Util.WordUtil(areCharsEqual)
+import qualified Data.Map.Strict as Map (Map, fromListWith, fromList, unionWith, findWithDefault)
+
+findBestWord :: [String] -> [String] -> Maybe String
+findBestWord searchDict inputDict =
+  let transformedWords = map (\candidate -> (candidate, getColorCountForDict candidate inputDict)) searchDict
+      h = head transformedWords
+  in if not $ null inputDict then Just $ fst $ foldl (\x y ->
+    if compareWords x y == LT then y else x
```

Comment:
If you provided an instance of `Ord` for the data type that represents the dictionaries, you could have then used minimum/maximum/sort and other useful standard library functions